src/Features/JsonPatch.SystemTextJson/src/Helpers/JsonUtilities.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace Microsoft.AspNetCore.JsonPatch.SystemTextJson.Helpers;
+
+internal static class JsonUtilities
+{
+    public static bool DeepEquals(object a, object b, JsonSerializerOptions serializerOptions)
+    {
+        return JsonObject.DeepEquals(
+            JsonSerializer.SerializeToNode(a, serializerOptions),
+            JsonSerializer.SerializeToNode(b, serializerOptions));
```

Comment:
```suggestion
    public static bool DeepEquals(object a, object b, JsonSerializerOptions serializerOptions)
    {
        using JsonDocument docA = JsonSerializer.SerializeToDocument(a, serializerOptions);
        using JsonDocument docB = JsonSerializer.SerializeToDocument(b, serializerOptions);

        return JsonElement.DeepEquals(docA.RootElement, docB.RootElement);
    }
```

This can also be optimized to check if either a or b are JsonElements and in that case `((JsonElement)a)` can be used instead of `JsonSerializer.SerializeToDocument(a, serializerOptions).RootElement`. This might be a common path since `JsonSerializer.Deserialize<object>(<foo>)` is used to deserialize incoming `Operation`/`Operation<T>` and that will return a JsonElement.